code/modules/research/stock_parts/stock_part_datum.dm

```diff
@@ -72,14 +72,129 @@ GLOBAL_LIST_INIT(stock_part_datums, generate_stock_part_datums())
 	physical_object_type = /obj/item/stock_parts/scanning_module
 	physical_object_base_type = /obj/item/stock_parts/scanning_module
 
-/datum/stock_part/scanning_module/tier2
+/datum/stock_part/scanning_module/adv
 	tier = 2
 	physical_object_type = /obj/item/stock_parts/scanning_module/adv
 
-/datum/stock_part/scanning_module/tier3
+/datum/stock_part/scanning_module/phasic
 	tier = 3
 	physical_object_type = /obj/item/stock_parts/scanning_module/phasic
 
-/datum/stock_part/scanning_module/tier4
+/datum/stock_part/scanning_module/triphasic
 	tier = 4
 	physical_object_type = /obj/item/stock_parts/scanning_module/triphasic
+
+/datum/stock_part/capacitor
+	tier = 1
+	physical_object_type = /obj/item/stock_parts/capacitor
+	physical_object_base_type = /obj/item/stock_parts/capacitor
+
+/datum/stock_part/capacitor/adv
+	tier = 2
+	physical_object_type = /obj/item/stock_parts/capacitor/adv
+
+/datum/stock_part/capacitor/super
+	tier = 3
+	physical_object_type = /obj/item/stock_parts/capacitor/super
+
+/datum/stock_part/capacitor/quadratic
+	tier = 4
+	physical_object_type = /obj/item/stock_parts/capacitor/quadratic
+
+/datum/stock_part/manipulator
+	tier = 1
+	physical_object_type = /obj/item/stock_parts/manipulator
+	physical_object_base_type = /obj/item/stock_parts/manipulator
+
+/datum/stock_part/manipulator/nano
+	tier = 2
+	physical_object_type = /obj/item/stock_parts/manipulator/nano
+
+/datum/stock_part/manipulator/pico
+	tier = 3
+	physical_object_type = /obj/item/stock_parts/manipulator/pico
+
+/datum/stock_part/manipulator/femto
+	tier = 4
+	physical_object_type = /obj/item/stock_parts/manipulator/femto
+
+/datum/stock_part/micro_laser
+	tier = 1
+	physical_object_type = /obj/item/stock_parts/micro_laser
+	physical_object_base_type = /obj/item/stock_parts/micro_laser
+
+/datum/stock_part/micro_laser/high
+	tier = 2
+	physical_object_type = /obj/item/stock_parts/micro_laser/high
+
+/datum/stock_part/micro_laser/ultra
+	tier = 3
+	physical_object_type = /obj/item/stock_parts/micro_laser/ultra
+
+/datum/stock_part/micro_laser/quadultra
+	tier = 4
+	physical_object_type = /obj/item/stock_parts/micro_laser/quadultra
+
+/datum/stock_part/matter_bin
+	tier = 1
+	physical_object_type = /obj/item/stock_parts/matter_bin
+	physical_object_base_type = /obj/item/stock_parts/matter_bin
+
+/datum/stock_part/matter_bin/adv
+	tier = 2
+	physical_object_type = /obj/item/stock_parts/matter_bin/adv
+
+/datum/stock_part/matter_bin/super
+	tier = 3
+	physical_object_type = /obj/item/stock_parts/matter_bin/super
+
+/datum/stock_part/matter_bin/bluespace
+	tier = 4
+	physical_object_type = /obj/item/stock_parts/matter_bin/bluespace
+
+/// Subspace stock parts
+
+/datum/stock_part/subspace/ansible
```

Comment:
don't make these share a subtype. there's no point, they don't share anything, and it breaks the code.

Reply:
Tests are failing cause it can't initialize datum/stock_part/subspace what do i do?

Reply:
I just have to rename them i guess